src/Core/WinSWCore/ServiceDescriptor.cs

```diff
@@ -95,6 +100,36 @@ public ServiceDescriptor(XmlDocument dom)
             this.dom = dom;
         }
 
+        private void ValidateAndLoadXmlSchema()
+        {
+            XmlReaderSettings settings = new XmlReaderSettings();
+
+            try
+            {
+                settings.Schemas.Add(null, BasePath + ".xsd");

```

Comment:
We can't ship the XSD as a physical file. Have a look around [Assembly.GetManifestResourceStream](https://docs.microsoft.com/dotnet/api/system.reflection.assembly.getmanifestresourcestream).